src/parse.c

```diff
@@ -4440,7 +4440,7 @@ void Parser::checkCstyleTypeSyntax(Loc loc, Type *t, int alt, Identifier *ident)
 
 Statement *Parser::parseStatement(int flags, const utf8_t** endPtr, Loc *pEndloc)
 {
-    Statement *s;
+    Statement *s = NULL;
```

Comment:
Oh, you need to delete both the `check(TOKsemicolon);` and the `break;` lines, because `Lerror` skips up until the next semicolon then skips the semicolon.  So if you have already skipped the semicolon it will completely skip the next statement.